controllers/topology/desired_state.go

```diff
@@ -715,3 +739,57 @@ func ownerReferenceTo(obj client.Object) *metav1.OwnerReference {
 		APIVersion: obj.GetObjectKind().GroupVersionKind().GroupVersion().String(),
 	}
 }
+
+// PrettyPrint returns a formatted JSON version of the object given.
+func PrettyPrint(v interface{}) string {
+	b, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		return err.Error()
+	}
+	return string(b)
+}
+
+func computeMachineHealthCheck(healthCheckTarget client.Object, selector *metav1.LabelSelector, clusterName string, check *clusterv1.MachineHealthCheckClass) *clusterv1.MachineHealthCheck {
+	// create a MachineHealthCheck with the spec given in the ClusterClass
+	return &clusterv1.MachineHealthCheck{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       clusterv1.MachineHealthCheck{}.Kind,
+			APIVersion: clusterv1.MachineHealthCheck{}.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Wondering if we should create MHS with pre-set owner references, thus avoiding orphaned objects.

Reply:
Current version of the code sets both the referenced object (controlplane or machineDeployment) and the Cluster as owners at MHC creation time.